### DISCUSSION TITLE
No matter how I set Size, it shows the smallest width and height, right

### DISCUSSION THREAD

**Discussion by yaoyuanmysql** [fyne-io/fyne#3775] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/25978114/228805296-03c6e6b4-b143-4070-ad18-7edbc56ce5ce.png)

code:
`
func main() {

	a := app.NewWithID("11")
	w := a.NewWindow("test")
	a.Settings().SetTheme(theme.LightTheme())

	re, err := fyne.LoadResourceFromPath("myapp.png")
	if err != nil {
		os.Exit(1)
	}

	entry := widget.NewEntry()
	entry.Resize(fyne.NewSize(50, entry.MinSize().Height)) // 设置宽度为200

	butn := widget.NewButton("xxx", func() {})
	butn.MinSize().Min(fyne.NewSize(800, 1200))

	logo := canvas.NewImageFromResource(re)
	logo.FillMode = canvas.ImageFillContain
	logo.SetMinSize(fyne.NewSize(50, 50))

	c := container.NewCenter(
		container.NewVBox(
			widget.NewLabelWithStyle("Welcome ", fyne.TextAlignCenter, fyne.TextStyle{Bold: true}), logo,
			entry,
			butn,
		))

	c.Resize(fyne.NewSize(50, 50))
	w.SetContent(c)
	w.Resize(fyne.NewSize(400, 300))
	w.ShowAndRun()
}
`

**Comment by Jacalz** [comment, marked as answer]
Have a look at https://developer.fyne.io/explore/layouts (especially the video at the bottom) and https://developer.fyne.io/faq/layout. Your VBox layout collapses all the content to its smallest vertical size (MinSize).

**Comment by yaoyuanmysql** [comment]
Although my problem has been solved, but the layout of fyne makes me feel that it is not flexible. I cannot set the size and position of a container.

**Reply by andydotxyz** [reply]
It is highly flexible - supporting rich and complex applications.
Some containers for specific layouts, and in that case it may feel inflexible because they do one job.
It is a little different to others and may be peculiar to people who like CSS layout. In Fyne you will need to think about the parent, not the child element, when arranging content. This actually supports much better flexibility and will adapt for a wide range of screens much more easily.

**Comment by yaoyuanmysql** [comment]
window.SetContent(
		container.NewCenter(
			container.NewGridWrap(
				fyne.NewSize(300, account.MinSize().Height),
				account,
				widget.NewLabel(""),
				container.NewCenter(
					container.NewGridWrap(fyne.NewSize(120, loginBtn.MinSize().Height),
						loginBtn),
				))))
